pkg/provider/azure_local_services.go

```diff
@@ -286,15 +286,71 @@ func (az *Cloud) getLocalServiceInfo(serviceName string) (*serviceInfo, bool) {
 	return data.(*serviceInfo), true
 }
 
+func getIPsFromEndpointSlice(es *discovery_v1.EndpointSlice) []string {
+	var currentIPs []string
+
+	if es != nil {
+		for _, endpoint := range es.Endpoints {
+
+			if endpoint.Conditions.Ready != nil && !*endpoint.Conditions.Ready {
+
+				continue
+			}
+
+			currentIPs = append(currentIPs, endpoint.Addresses...)
+
+		}
+	}
+	return currentIPs
+}
+
+func (az *Cloud) createOrUpdateBackendPoolForES(es *discovery_v1.EndpointSlice, si *serviceInfo) {
+	currentIPs := getIPsFromEndpointSlice(es)
+
+	isIPv6 := es.AddressType == discovery_v1.AddressTypeIPv6
+	poolName := az.getBackendPoolNameForEndpointSlice(es, isIPv6)
+
+	bp := network.BackendAddressPool{
+		Name:                               &poolName,
+		BackendAddressPoolPropertiesFormat: &network.BackendAddressPoolPropertiesFormat{},
+	}
+
+	az.addIPAddressesToBackendPool(&bp, currentIPs)
+
+	klog.V(4).Infof("Creating or updating backend pool - %s", poolName)
+	az.LoadBalancerClient.CreateOrUpdateBackendPools(context.Background(), az.ResourceGroup, si.lbName, poolName, bp, pointer.StringDeref(bp.Etag, ""))
+}
+
 // setUpEndpointSlicesInformer creates an informer for EndpointSlices of local services.
 // It watches the update events and send backend pool update operations to the batch updater.
+// TODO (anujbansal): Handle LB rule update
 func (az *Cloud) setUpEndpointSlicesInformer(informerFactory informers.SharedInformerFactory) {
 	endpointSlicesInformer := informerFactory.Discovery().V1().EndpointSlices().Informer()
 	_, _ = endpointSlicesInformer.AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				es := obj.(*discovery_v1.EndpointSlice)
 				az.endpointSlicesCache.Store(strings.ToLower(fmt.Sprintf("%s/%s", es.Namespace, es.Name)), es)
+
+				if az.LoadBalancerBackendPoolConfigurationType != consts.LoadBalancerBackendPoolConfigurationTypePodIP {
+					klog.V(4).Infof("Backend Pool configuration is %s, skipping backend pool creation", az.LoadBalancerBackendPoolConfigurationType)
+					return
+				}
+
+				svcName := getServiceNameOfEndpointSlice(es)
+				if svcName == "" {
+					klog.V(4).Infof("EndpointSlice %s/%s does not have service name label, skip creating load balancer backend pool", es.Namespace, es.Name)
+					return
+				}
+
+				key := strings.ToLower(fmt.Sprintf("%s/%s", es.Namespace, svcName))
+				si, found := az.getLocalServiceInfo(key)
```

Comment:
Entry is added into az.localServiceNameToServiceInfoMap only "if az.useMultipleStandardLoadBalancers() && isLocalService(service)". For CLB, useMultipleStandardLoadBalancers() is false. Wondering how this works for updating backendpool for CLB. Let's discuss offline.